src/language-js/needs-parens.js

```diff
@@ -386,31 +386,8 @@ function needsParens(path, options) {
       }
 
     case "SequenceExpression":
-      switch (parent.type) {
-        case "ReturnStatement":
-          return false;
-
-        case "ForStatement":
-          // Although parentheses wouldn't hurt around sequence
-          // expressions in the head of for loops, traditional style
-          // dictates that e.g. i++, j++ should not be wrapped with
-          // parentheses.
-          return false;
-
-        case "ExpressionStatement":
-          return name !== "expression";
-
-        case "ArrowFunctionExpression":
-          // We do need parentheses, but SequenceExpressions are handled
-          // specially when printing bodies of arrow functions.
```

Comment:
How and why exactly are they handled specially? Did you investigate these comments @sosukesuzuki?

Reply:
No, I haven't investigate these comments. Sorry, I thought you know. I will investigate.

Reply:
This was added at #2388